### ISSUE TITLE
Self-hosted Gitlab CE 

### ISSUE THREAD

**Issue by thedanielhanke** [buunguyen/octotree#276]
Hi,

i tested it on several osx and linux chrome (v49, plugon version 2.0.9) instances, to get your awesome plugin to work with gitlab.

so i added the url including protocol, and confirmed the plugin access to my website. 
then on my gitlab sites, it asks me for the token. so far, everything fine.

entering a token, valid or not, always leads to url-calls like

(space after :// for formatting) 
jquery.js:8630 GET https:// git.domain.tld/api/v3/repos/`repository/name`?_=1458540256084 404 ()

whereas `repos/ID` would be the correct url, based on the routes-config of GL (correct me if i am wrong here, but when i enter the project_id that urls work)

```js
$('#project_id')
[<input type=​"hidden" name=​"project_id" id=​"project_id" value=​"61">​]
```

the project_id is found.

my gitlab is a unedited latest 8.5 version. 

Any information i can give, any ideas you have?
thanks a lot.

**Comment by thedanielhanke** [comment]
okay. i just built it on my own - after solving some JS-Issues i had(encoding-issues due to hotkeys) - with a version number of 2.0.10, and now it works. meanwhile i tested it on chrome and firefox with cuirrent 2.0.9, same results as in chrome (404..). 

no clue if that helps or leads to confusion ;)

**Comment by buunguyen** [comment]
What changes did you make from 2.0.9 to your own 2.0.10?

**Comment by catesandrew** [comment]
I got it to work but had to [comment out](https://github.com/buunguyen/octotree/blob/master/gulpfile.babel.js#L197) the gulp babel command in the pipeline. for whatever reason, it was garbling the octotreee.js in a weird manner. i also added my private gitlab urls, but that goes without saying hopefully. comment that line out and then gulp build, should work.

**Comment by thedanielhanke** [comment]
@catesandrew indeed, commenting out that line OR[ the `@ifdef`  lines](https://github.com/buunguyen/octotree/blob/master/src%2Fconstants.js#L23) does it.


@buunguyen **TL;DR** it only works, when i add the gitlab-url in manifest.permissions. 

------

 my changes are version-number bump, the commenting mentioned, and also i added my server to default permissions.

i now also removed that line again, rebuild/reinstalled, and it stopped to work,  calling the wrong urls again. the format of my urls does match. for testing purposes i also tested 

- https://domain.tld
- https://domain.tld/
- https://domain.tld/*   <-  like in manifest.permissions

always the same : Save config -> Prompt for Access -> Agreee! -> goto domain.tld -> 404 via octotree.

testet it with my macbook in osx with current chrome too. same result.
i dont have any networking constraints in place on any machine.

**Comment by ulrichSchreiner** [comment]
Hi,

is there anything one can do to make this work in the official version?

**Comment by buunguyen** [comment]
@thedanielhanke I suppose removing all those `@ifdef SAFARI` will address the issue? Would you mind testing and submitting a PR if it does fix the issue. Thanks a bunch.

**Comment by thedanielhanke** [comment]
@buunguyen  sorry i missed that comment. will do this week. cheers

**Comment by buunguyen** [comment]
@thedanielhanke any update on this?

**Comment by buunguyen** [comment]
I commented out the conditional short key statements and released as part of 2.0.10. Please double-check and let me know if the bug still exists.
